### PR TITLE
Add Caltech101, Caltech256, and CelebA

### PR DESCRIPTION
**Pull request by bryanhe**

### CODE REVIEW

File: torchvision/datasets/utils.py [hunk 1]
```diff
@@ -127,3 +127,58 @@ def list_files(root, suffix, prefix=False):
         files = [os.path.join(root, d) for d in files]
 
     return files
+
+
+def download_file_from_google_drive(id, root, filename=None, md5=None):
+    """Download a Google Drive file from  and place it in root.
+
+    Args:
+        url (str): URL to download file from
```
Comment (fmassa) [inline torchvision/datasets/utils.py]: the doc mention `url`, while the code is written `id` (which is by the way a Python method which you are shadowing in this function).  Could you make it consistent, by either fixing the doc or changing the name in the function argument? I'd prefer to change from `id` to something else

**Comment by bryan-he** [comment]
I think all of the issues should be resolved now.

I don't think many papers only use Caltech101/256, but I've seen it used for domain adaptation experiments in somewhat recent papers, such as http://papers.nips.cc/paper/6757-learning-multiple-tasks-with-deep-relationship-networks (2017), along with a few papers that it cites.

Do you think some of those (Office-Home and ImageCLEF) are worth including?

File: torchvision/datasets/celeba.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+        with open(os.path.join(self.root, self.base_folder, "list_landmarks_align_celeba.txt"), "r") as f:
+            self.landmarks_align = pandas.read_csv(f, delim_whitespace=True, header=1)
+
+        with open(os.path.join(self.root, self.base_folder, "list_attr_celeba.txt"), "r") as f:
+            self.attr = pandas.read_csv(f, delim_whitespace=True, header=1)
+
+        mask = (splits[1] == split)
+        self.filename = splits[mask].index.values
+        self.identity = torch.LongTensor(self.identity[mask].values)
```
Comment (fmassa) [inline torchvision/datasets/celeba.py]: `torch.LongTensor` is the legacy constructor, could you use instead `torch.as_tensor`? If you want to force it to be `int64`, you can do `torch.as_tensor(..., dtype=torch.int64)`.

**Comment by fmassa** [comment]
I haven't myself seen / used Office-Home and ImageCLEF be used much, but if you think it would be valuable to have them in torchvision, if you could send a PR adding them it would be great

**Comment by bryanhe** [comment]
Last comment is resolved now.

I'll think about whether those two are worth including.
I was also going to add in a few other datasets (DeepFashion and ADE2K are fairly popular, so I assume they're worth adding?). Does it make sense to PR each on separately, or just block them together?

**Comment by fmassa** [comment]
Separate PRs are better I think.

Thanks!

**Pull request merged**